inc/search.class.php

```diff
@@ -1264,6 +1264,11 @@ static function constructData(array &$data, $onlycount = false) {
          }
       }
 
+      if (!$result) {
+         echo $DBread->error();
+         return false;
+      }
```

Comment:
reverse the if ($result) {...} else {...} found below, for readability, and return a boolean instead of nothing